sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -546,7 +547,8 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
 case class WithFields(
     structExpr: Expression,
     names: Seq[String],
-    valExprs: Seq[Expression]) extends Unevaluable {
+    valExprs: Seq[Expression],
+    sortColumns: Boolean = false) extends Unevaluable {
```

Comment:
How about defining `toString` for not displaying this value in explain output?

Reply:
nit: `sortColumns` -> `sortOutputColumns`?

Reply:
Changed `sortColumns` to `sortOutputColumns`. I'm not sure we want to hide `sortColumns`?

Reply:
I thought this param is not related to `withField`, so I left the comment. But, either is okay.
```
scala> val df = sql("SELECT named_struct('a', 1, 'b', 2) struct_col")
scala> df.select($"struct_col".withField("c", lit(3))).explain(true)
== Analyzed Logical Plan ==
with_fields(struct_col, 3): struct<a:int,b:int,c:int>
Project [with_fields(struct_col#0, c, 3, false) AS with_fields(struct_col, 3)#4]
                                         ^^^^^
+- Project [named_struct(a, 1, b, 2) AS struct_col#0]
   +- OneRowRelation
```

Reply:
I'm not certain if we want to show it or not. Let's keep it as is and see what others think.